lib/datadog/appsec/security_engine/engine.rb

```diff
@@ -129,7 +133,7 @@ def load_default_config(telemetry:)
           config['exclusions'] ||= AppSec::Processor::RuleLoader.load_exclusions(ip_passlist: @default_ip_passlist)
 
           diagnostics = @waf_builder.add_or_update_config(config, path: DEFAULT_RULES_CONFIG_PATH)
-          @ruleset_version = diagnostics['ruleset_version']
+          @builder_ruleset_version = diagnostics['ruleset_version']
```

Comment:
This is very unobvious move. In the code above there is no clue how the `@builder_ruleset_version` was set, even tho we return result from which we can retrieve that data and now it become significant to spot as the issue.

Reply:
I improved it a little in [43d830a](https://github.com/DataDog/dd-trace-rb/pull/4839/commits/43d830a36bea160e5d7e58eb770f0bdd5511dcfd). But we still have to keep the ruleset version that is in the builder, but not in the handle (since it wasn't reconfigured yet).

One option would be to do it in `WAF::HandleBuilder` in `libddwaf-rb`, what do you think?